sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSqlSchema.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.schema;
+
+import com.google.inject.Inject;
+import org.apache.calcite.schema.Schema;
+
+/**
+ * The schema for druid tables to be accessible via sql.
+ */
+class DruidSqlSchema implements DruidCalciteSchema
```

Comment:
We now have classes DruidSchema, DruidSqlSchema, and DruidCalciteSchema. Their relationships aren't very clear from the names. How about naming them DruidSchema, DruidSchemaProvider, and SchemaProvider respectively?

Reply:
I don't like `Provider` because that's a guice thing and I'd expect it to have a method that returns an object, but this interface returns 2 things and it doesn't implement Provider<T>

How about
* `<Druid>Schema` which provides the implementation that Calcite needs to get tables
* `CalciteSubSchema` which is an interface containing a name and a `Schema` so that it can be registered to a Root Schema(`SchemaPlus`)
* `<System>SubSchema` the implementation of the `CalciteSubSchema` interface for appropriate `<System>Schema`

Reply:
So there'd be both a SystemSchema and SystemSubSchema (and the SystemSubSchema would be how you get a SystemSchema)?

Who would have thought this would be the hard part.

How about DruidSchema, NamedDruidSchema, and NamedSchema?

Reply:
lol - alright I like that suggestion. Thanks!

Reply:
:|